x/genutil/client/cli/migrate_test.go

```diff
@@ -52,13 +68,52 @@ func TestMigrateGenesis(t *testing.T) {
 	defer cleanup()
 
 	// Reject if we dont' have the right parameters or genesis does not exists
-	require.Error(t, MigrateGenesisCmd(ctx, cdc).RunE(nil, []string{target, genesisPath}))
+	require.Error(t, genutilcli.MigrateGenesisCmd(ctx, cdc).RunE(nil, []string{target, genesisPath}))
 
 	// Noop migration with minimal genesis
 	emptyGenesis := []byte(`{"chain_id":"test","app_state":{}}`)
 	err = ioutil.WriteFile(genesisPath, emptyGenesis, 0600)
 	require.Nil(t, err)
 	cmd := setupCmd("", "test2")
-	require.NoError(t, MigrateGenesisCmd(ctx, cdc).RunE(cmd, []string{target, genesisPath}))
+	require.NoError(t, genutilcli.MigrateGenesisCmd(ctx, cdc).RunE(cmd, []string{target, genesisPath}))
 	// Every migration function shuold tests its own module separately
 }
+
+func TestMigrateCommercioGenesisData(t *testing.T) {
+	home, cleanup := tests.NewTestCaseDir(t)
+	t.Cleanup(cleanup)
+	viper.Set(cli.HomeFlag, home)
+	viper.Set(flags.FlagName, "moniker")
+	logger := log.NewNopLogger()
+	cfg, err := tcmd.ParseConfig()
+	require.Nil(t, err)
+	ctx := server.NewContext(cfg, logger)
+	cdc := makeCodec()
+
+	genesisPath := filepath.Join("testdata", "commercio-genesis.json")
+	require.NoError(t, err)
+	target := "v0.39"
+
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount("did:com:", "did:com:pub")
+	config.Seal()
+
+	// Migration with minimal genesis
```

Comment:
Okay, this is a nice sanity check. Using a real v0.38.x genesis file and migrating it and loading it.

It also makes it easy enough to swap out for a larger genesis file if we have a more complex testcase we need to validate, so this is super-useful setup code.